### PR TITLE
Handle IPatternCaseClauseOperation in control flow graph

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@333fred, @mavasani, @cston, @dotnet/roslyn-compiler, @dotnet/analyzer-ioperation Please review.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ISwitchOperation.cs [hunk 1]
```diff
@@ -1963,6 +2150,342 @@ void M(int input1, MyClass input2)
...
+Block[B4] - Exit
+    Predecessors: [B2] [B3]
+    Statements (0)
+";
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void SwitchFlow_30()
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ISwitchOperation.cs]: Can we get a few more guard tests? I'd like to see something for an invalid guard, as well as what would happen if we have a throw expression in a guard (ie `when ((object1 ?? throw new Exception()) == true)` or similar).

**Comment by 333fred** [comment]
Done review (iteration 1)

**Review by mavasani — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ISwitchOperation.cs (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ISwitchOperation.cs]: > Can we get a few more guard tests? I'd like to see something for an invalid guard, as well as what would happen if we have a throw expression in a guard (ie when ((object1 ?? throw new Exception()) == true) or similar).

I do not think these scenarios are particularly interesting, but I will add some tests in the next PR. 

---
In reply to: [184755922](https://github.com/dotnet/roslyn/pull/26440#discussion_r184755922) [](ancestors = 184755922)

**Review by 333fred — APPROVED** [review]
@AlekseyTs confirmed the tests revealed no new bugs, so they're fine in a follow up PR.

**Pull request merged**